requirements.txt

```diff
@@ -1,7 +1,15 @@
+amqp==5.1.1
```

Comment:
Is there anyway to know which of these our direct dependencies?
We usually only pin our direct dependencies and if we are interested in pinning our indirect dependencies we use something like pip-tools or something similar to pin them.

Reply:
Thank you, I have resolved this. I have added the pre-commit stage of generating the requirements.txt from the direct dependancies in requirements.in.